service/resource/resourcegroup/resource.go

```diff
@@ -0,0 +1,298 @@
+package resourcegroup
+
+import (
+	azureresource "github.com/Azure/azure-sdk-for-go/arm/resources/resources"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/giantswarm/azuretpr"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/operatorkit/framework"
+
+	"github.com/giantswarm/azure-operator/client"
+	"github.com/giantswarm/azure-operator/service/key"
+)
+
+const (
+	// Name is the identifier of the resource.
+	Name = "resourcegroup"
+
+	clusterIDTag  = "ClusterID"
+	customerIDTag = "CustomerID"
+	managedBy     = "azure-operator"
+)
+
+type Config struct {
+	// Dependencies.
+
+	AzureConfig *client.AzureConfig
+	Logger      micrologger.Logger
+}
+
+// DefaultConfig provides a default configuration to create a new service by
+// best effort.
+func DefaultConfig() Config {
+	return Config{
+		// Dependencies.
+		AzureConfig: nil,
+		Logger:      nil,
+	}
+}
+
+type Resource struct {
+	// Dependencies.
+
+	azureConfig *client.AzureConfig
+	logger      micrologger.Logger
+}
+
+// New creates a new configured resource group resource.
+func New(config Config) (*Resource, error) {
+	// Dependencies.
+	if config.AzureConfig == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.AzureConfig must not be empty.")
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.Logger must not be empty.")
+	}
+
+	newService := &Resource{
+		azureConfig: config.AzureConfig,
+		logger: config.Logger.With(
+			"resource", Name,
+		),
+	}
+
+	return newService, nil
+}
+
+// GetCurrentState gets the resource group for this cluster from the Azure API.
+func (r *Resource) GetCurrentState(obj interface{}) (interface{}, error) {
+	customObject, err := toCustomObject(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	var resourceGroup azureresource.Group
+	{
+		groupsClient, err := r.getGroupsClient()
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		resourceGroup, err = groupsClient.Get(key.ClusterID(customObject))
+		if err != nil {
+			if client.ResponseWasNotFound(resourceGroup.Response) {
+				// Fall through.
+				return nil, nil
+			}
+
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	return &resourceGroup, nil
+}
+
+// GetDesiredState returns the desired resource group for this cluster.
+func (r *Resource) GetDesiredState(obj interface{}) (interface{}, error) {
+	customObject, err := toCustomObject(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	r.logger.Log("cluster", key.ClusterID(customObject), "debug", "computing the new resource group")
```

Comment:
FYI I find these additional logs quite spammy and redundant when the log resource is used for wrapping. IN the draughtsman operator I removed this logging already and I would remove it in other operators as well.

Reply:
Cool, I'll remove here too. Good to standardise across operators.

Reply:
Done.